modules/legacy-geo/src/main/java/org/elasticsearch/legacygeo/parsers/GeoWKTParser.java

```diff
@@ -102,26 +102,17 @@ private GeoWKTParser() {}
                 throw new ElasticsearchParseException("Expected geometry type [{}] but found [{}]", shapeType, type);
             }
         }
-        switch (type) {
-            case POINT:
-                return parsePoint(stream, ignoreZValue, coerce);
-            case MULTIPOINT:
-                return parseMultiPoint(stream, ignoreZValue, coerce);
-            case LINESTRING:
-                return parseLine(stream, ignoreZValue, coerce);
-            case MULTILINESTRING:
-                return parseMultiLine(stream, ignoreZValue, coerce);
-            case POLYGON:
-                return parsePolygon(stream, ignoreZValue, coerce);
-            case MULTIPOLYGON:
-                return parseMultiPolygon(stream, ignoreZValue, coerce);
-            case ENVELOPE:
-                return parseBBox(stream);
-            case GEOMETRYCOLLECTION:
-                return parseGeometryCollection(stream, ignoreZValue, coerce);
-            default:
-                throw new IllegalArgumentException("Unknown geometry type: " + type);
-        }
+        return switch (type) {
+            case POINT -> parsePoint(stream, ignoreZValue, coerce);
+            case MULTIPOINT -> parseMultiPoint(stream, ignoreZValue, coerce);
+            case LINESTRING -> parseLine(stream, ignoreZValue, coerce);
+            case MULTILINESTRING -> parseMultiLine(stream, ignoreZValue, coerce);
+            case POLYGON -> parsePolygon(stream, ignoreZValue, coerce);
+            case MULTIPOLYGON -> parseMultiPolygon(stream, ignoreZValue, coerce);
+            case ENVELOPE -> parseBBox(stream);
+            case GEOMETRYCOLLECTION -> parseGeometryCollection(stream, ignoreZValue, coerce);
+            default -> throw new IllegalArgumentException("Unknown geometry type: " + type);
```

Comment:
This switch is missing `CIRCLE`. Can we find out if that's correct? If it's not correct, then we need to add `CIRCLE` and then we can remove the `default`.

Reply:
Looking at #27417 it seems that `CIRCLE` is explicitly not supported by the WKT parser.